utils_cv/detection/plot.py

```diff
@@ -65,26 +66,46 @@ def plot_boxes(
         draw = ImageDraw.Draw(im)
         font = get_font(size=plot_settings.text_size)
 
+        # get color map and convert to list of integer tuples
+        if not (plot_settings.rect_color and plot_settings.text_color):
+            colors = matplotlib.cm.get_cmap('tab20').colors
+            colors = np.floor(np.array(colors) * 255).astype('int')
+            colors = tuple(map(tuple, colors))
+
         for bbox in bboxes:
             # do not draw background bounding boxes
             if hasattr(bbox, "label_idx") and bbox.label_idx == 0:
                 continue
 
-            box = [(bbox.left, bbox.top), (bbox.right, bbox.bottom)]
+            # show detection score in rectangle label
+            bbox_text = bbox.label_name
+            if type(bbox) is DetectionBbox:
+                bbox_text += " ({:0.2f})".format(bbox.score)
+
+            # pick rectangle and text color
+            if plot_settings.rect_color and plot_settings.text_color:
```

Comment:
i think this functionality can be part of the PlotSettings class. `plot_setting.get_next_color()` or something like that

Reply:
this would also be conceptually a bit nicer because the logic of picking colors/ etc will be kept under `PlotSettings`, and the `plot_boxes` function won't have to deal with this.

Reply:
Added get_color() function to the PlotSettings class. Does indeed look nicer